broker/recreate.go

```diff
@@ -26,7 +26,11 @@ func (bkr *Broker) Recreate(instanceID string, details brokerapi.ProvisionDetail
 
 	clusterState := bkr.initClusterStateFromRecreationData(recreationData)
 	go func() {
-		features := bkr.clusterFeaturesFromProvisionDetails(details)
+		features, err := bkr.clusterFeaturesFromProvisionDetails(details)
```

Comment:
This should also be in the sync part (outside `go func(){}`). Also I suggest separating the verification from the initalization of the features.